python/ray/data/datasource/file_based_datasource.py

```diff
@@ -355,6 +355,13 @@ def __init__(
             path_to_size = dict(zip(self._paths, self._file_sizes))
             self._paths = self._partition_filter(self._paths)
             self._file_sizes = [path_to_size[p] for p in self._paths]
+            if len(self._paths) == 0:
+                raise ValueError(
+                    "Not found any input file to read from. Please double "
+                    "check the input files are having expected extension(s): "
```

Comment:
It is not necessarily doing an extension filtering (it can be just partition filtering yields empty paths) so we should not just point to extension.

Reply:
@jianoaix - thanks for pointing it out. I agree with your point, so changed the error message. I am happy to look into a better API for path filtering as followup. Right now I think the error message is pretty confusing `ValueError: number sections must be larger than 0.` and worth to fix. WDYT?

Reply:
Yes, sorry I wasn't meaning to get the API name right in this PR.

For the message itself, if more detailed/specific information wanted, maybe use `isinstance()` to dispatch PartitionFilter and ExtensionFilter?

Reply:
> maybe use isinstance() to dispatch PartitionFilter and ExtensionFilter?

hmm for `PathPartitionFilter` it seems a little hard to extract a message - we support an arbitrary `filter_fn` inside it. I changed the error message to be:

```
"Not found any input file to read from. Please double check
'partition_filter' field is set properly."
```